### PR TITLE
Pr 8341 and payload len fix/v1

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
https://redmine.openinfosecfoundation.org/issues/5379
https://redmine.openinfosecfoundation.org/issues/5693

suricata-verify-pr: 1055

### CODE REVIEW

**Comment by inashivb** [comment]
Wait a second. This is incomplete. The OP had reported this issue with TCP as well so that needs test and fixing too..

File: src/detect-engine-event.c [hunk 1]
```diff
@@ -160,6 +160,9 @@ static DetectEngineEventData *DetectEngineEventParse (const char *rawstr)
 
     if (de->event == STREAM_REASSEMBLY_OVERLAP_DIFFERENT_DATA) {
         StreamTcpReassembleConfigEnableOverlapCheck();
+    } else if (de->event == UDP_HLEN_INVALID) {
+        SCLogWarning("Rule uses decode-event \"udp.hlen_invalid\" which will be deprecated in "
```
Comment (lukashino) [inline src/detect-engine-event.c:164]: Just a note: Suricata 8.0 was not agreed upon, I just put it there as a safe milestone when to remove the keyword. So @victorjulien needs to confirm this. Alternatively, the keyword can be used for a case that Shivani fixed.

**Review by lukashino — APPROVED** [review]
Looks good to me

File: src/decode-udp.c [hunk 2]
```diff
@@ -56,16 +56,16 @@ static int DecodeUDPPacket(ThreadVars *t, Packet *p, const uint8_t *pkt, uint16_
         return -1;
     }
 
-    if (unlikely(len != UDP_GET_LEN(p))) {
-        ENGINE_SET_INVALID_EVENT(p, UDP_HLEN_INVALID);
+    if (unlikely(UDP_GET_LEN(p) < UDP_HEADER_LEN)) {
+        ENGINE_SET_INVALID_EVENT(p, UDP_PKT_TOO_SMALL);
```
Comment (victorjulien) [inline src/decode-udp.c:60]: the issue here is that the length field isn't correct, not that the packet is too small?

Comment (inashivb) [reply src/decode-udp.c:60]: Yes. HLEN_INVALID also did not seem to fit.. I think in my head, I went with this bc the length received from the pkt is smaller than 8bytes..
New error code?

Comment (victorjulien) [reply src/decode-udp.c:60]: The pkt too small class of events is meant to indicate the input data is too small, not that a length field is wrong

Comment (inashivb) [reply src/decode-udp.c:60]: Indeed. That was an incorrect event. So is UDP_HLEN_INVALID and UDP_HLEN_TOO_SMALL, would you agree? I'm thinking these indicate that the packet came with an invalid or too small header length. This event is however about the entire length of the packet.

Should we make a new event for this case?

Comment (victorjulien) [reply src/decode-udp.c:60]: yeah I think so, something like `UDP_LEN_INVALID` or something

**Pull request closed without merging**